zebra-chain/src/orchard/note.rs

```diff
@@ -57,14 +62,16 @@ impl From<Nullifier> for Rho {
 }
 
 impl Rho {
-    pub fn new<T>(csprng: &mut T) -> Self
+    pub fn new<T>(csprng: &mut T) -> Result<Self, RandError>
     where
         T: RngCore + CryptoRng,
     {
         let mut bytes = [0u8; 32];
-        csprng.fill_bytes(&mut bytes);
+        csprng
+            .try_fill_bytes(&mut bytes)
+            .map_err(|_| RandError::FillBytes)?;
 
-        Self(extract_p(pallas::Point::from_bytes(&bytes).unwrap()))
+        Ok(Self(extract_p(pallas::Point::from_bytes(&bytes).unwrap())))
```

Comment:
I don't think we should panic here: instead, we should handle the identity point by returning an error:
https://docs.rs/pasta_curves/latest/pasta_curves/struct.Ep.html#method.from_bytes

Reply:
Done here: https://github.com/ZcashFoundation/zebra/pull/6385/commits/47a3adb948b751f51fd459e9317028ee7f25512b